runtime/doc/lua.txt

```diff
@@ -827,6 +827,62 @@ vim.schedule({callback})				*vim.schedule()*
         Schedules {callback} to be invoked soon by the main event-loop. Useful
         to avoid |textlock| or other temporary restrictions.
 
+
+vim.defer_fn({fn}, {timeout})                                    *vim.defer_fn*
+    Defers calling {fn} until {timeout} ms passes.  Use to do a one-shot timer
+    that calls {fn}.
+
+    Parameters: ~
+        {fn}        Callback to call once {timeout} expires
+        {timeout}   Time in ms to wait before calling {fn}
+
+    Returns: ~
+        |vim.loop|.new_timer() object
+
+vim.wait({time}, {callback} [, {interval}])                         *vim.wait()*
+        Wait for {time} in milliseconds until {callback} returns `true`.
+
+        Executes {callback} immediately and at approximately {interval}
+        milliseconds (default 200). Nvim still processes other events during
+        this time.
+
+
+    Returns: ~
+        If {callback} returns `true` during the {time}:
+            `true, nil`
+
+        If {callback} never returns `true` during the {time}:
+            `false, -1`
+
+        If {callback} is interrupted during the {time}:
+            `false, -2`
```

Comment:
What about using 0, -1, -2 for the same as vim.wait instead of returning multiple values?
I think it's less surprising that you have both vim.wait and the return interface.

https://github.com/neovim/neovim/pull/10382/files#diff-bea881dfa9626bab7141337b0fcdb23eR8877-R8879

Reply:
The reason I do not want the values to be exactly the same (although I'd be fine having it return `true, 0` instead of `true, nil` is because `0` is "truthy" in lua.

I think it is much more "lua-ish" to make it return in the style of `pcall`, where you have it return `success, code` style.